lib/azure/armrest/model/base_model.rb

```diff
@@ -62,7 +64,18 @@ def initialize(json)
           @json = json
         end
 
-        @ostruct = OpenStruct.new(hash)
+        ostruct = OpenStruct.new(hash)
+        keys = ostruct.to_h.keys
+
+        # If the method already exists then create an "_alias" method for it.
+        methods.each do |m|
+          if keys.include?(m)
+            ostruct.class.instance_eval{ alias_method "_#{m}", m }
+            instance_eval{ define_singleton_method("_#{m}"){ ostruct[m] } }
```

Comment:
It creates a new method to the hosting object, which is OK, but all our other delegating methods are defined in the delegated object.

My suggestion is continue to define all delegating methods in the delegated object in `__setobj__` method. The implementation is very simple, only two lines change. See my example:

```
      def __setobj__(obj)
        host_methods = methods
        excl_list = self.class.send(:excl_list)
        obj.methods(false).each{ |m|
          if m.to_s[-1] != '=' && !excl_list.include?(m.to_s) # Must deal with nested models
            res = obj.send(m)
            if res.is_a?(Array)
              newval = res.map { |elem| elem.is_a?(Hash) ? @embedModel.new(elem) : elem }
              obj.send("#{m}=", newval)
            elsif res.is_a?(Hash)
              obj.send("#{m}=", @embedModel.new(res))
            end
          end

          snake = m.to_s.gsub(/(.)([A-Z])/,'\1_\2').downcase.to_sym
          snake = snake.to_s.prepend('_').to_sym if host_methods.include?(snake)

          begin
            obj.instance_eval("alias #{snake} #{m}; undef :#{m}") unless snake == m
          rescue SyntaxError
            next
          end
        }
      end
```

This way no need to modify the `initialize` method.